src/book/03-guides/01-main-codebase/02-maintaining-subsystems.mdx

```diff
@@ -85,9 +124,37 @@ After updating the Visual Mesh in the NUbots repository, it should be tested bef
 If you would like to see the Visual Mesh output in NUsight, you will need to log the data and run it back in NUsight using DataPlayback, since the data is too large to send over a network. Use the steps in the [DataLogging and DataPlayback guide](/guides/main/data-recording-playback) to record and playback data. Adjust the instructions for our purpose using the following hints:
 
 - In step 1 of Recording Data, use the [`visualmesh`](https://github.com/NUbots/NUbots/blob/main/roles/visualmesh.role) role to record the data.
-- In step 2 of Recording Data and step 4 of Playing Back Data, set `message.output.CompressedImage` to `true` and add `message.vision.VisualMesh: true` in both [`DataLogging.yaml](https://github.com/NUbots/NUbots/blob/main/module/support/logging/DataLogging/data/config/DataLogging.yaml) and [`DataPlayback.yaml](https://github.com/NUbots/NUbots/blob/main/module/support/logging/DataPlayback/data/config/DataPlayback.yaml).
+- In step 2 of Recording Data and step 4 of Playing Back Data, set `message.output.CompressedImage` to `true` and add `message.vision.VisualMesh: true` in both [`DataLogging.yaml`](https://github.com/NUbots/NUbots/blob/main/module/support/logging/DataLogging/data/config/DataLogging.yaml) and [`DataPlayback.yaml`](https://github.com/NUbots/NUbots/blob/main/module/support/logging/DataPlayback/data/config/DataPlayback.yaml).
 - In steps 1, 2 and 5 of Playing Back Data, use the [`playback`](https://github.com/NUbots/NUbots/blob/main/roles/playback.role) role to playback the data, without changes.
 
+### Image Compressors
+
+In order to send data to NUsight via the network, images need to be compressed. Depending on the environment different compressors can be used. [TurboJPEG](https://github.com/libjpeg-turbo/libjpeg-turbo) is the compressor that can be used on all platforms and is used primarily when submitting docker images for virtual competitions as the processor type cannot be guaranteed. [VAAPI](https://01.org/linuxmedia/vaapi) is the preferred compressor for devices running on Intel CPUs, which support the framework.
+
+The configuration for either of these belongs in `ImageCompressor.yaml` and is as follows
+
+#### TurboJPEG
+
+```bash
```

Comment:
Should these code blocks be YAML?

```suggestion
```yaml
```

Reply:
Didn't even realise that was an option, will change

Reply:
Yeah we should probably document the list of supported languages on the NUbook page.

We supported everything here: https://github.com/FormidableLabs/prism-react-renderer/blob/v1.2.1/src/vendor/prism/includeLangs.js + Cmake and Docker:

https://github.com/NUbots/NUbook/blob/91b42679102f592ac611855b2c3b8493e36235aa/src/components/markdown/pre-code.jsx#L11-L13